inventory_management_system_api/repositories/setting.py

```diff
@@ -0,0 +1,107 @@
+"""
+Module for providing a repository for managing settings in a MongoDB database.
+"""
+
+import logging
+from typing import Optional, Type, TypeVar
+
+from pymongo.client_session import ClientSession
+from pymongo.collection import Collection
+
+from inventory_management_system_api.core.database import DatabaseDep
+from inventory_management_system_api.models.setting import SettingInBase, SettingOutBase, SparesDefinitionOut
+
+logger = logging.getLogger()
+
+# Template types for models inheriting from SettingIn/OutBase so this repo can be used generically for multiple settings
+SettingInBaseT = TypeVar("SettingInBaseT", bound=SettingInBase)
+SettingOutBaseT = TypeVar("SettingOutBaseT", bound=SettingOutBase)
```

Comment:
I think having the union here would actually mean it can be any subtype of SettingOutBase or SettingInBase (https://stackoverflow.com/questions/59933946/difference-between-typevart-a-b-and-typevart-bound-uniona-b). I don't think there is a way to have them coupled like that unfortunately.

![image](https://github.com/user-attachments/assets/4e263047-f587-4b52-a73d-c5acf8fb5189)